Functional enrichment analysis of DE genes
==========================================

Now, one would like to carry out pathway analysis which gives an idea of what specific processes/categories these transcripts are enriched in. Gene ontology (GO) is an oft-used method to assign the differentially expressed genes from a RNA-Seq workflow to a process/category and give context to a set of transcripts. 

The mission of the GO consortium is to develop a comprehensive, computational model of biological systems, ranging from the molecular level to the organism level, across the multiplicity of species in the tree of life (ref)
The three main categories of GO are Biological Process (BP), Cellular Component (CC) and Molecular Function (MF). To understand more of these sub-categories, please refer to the paper. Most of the tools that are used to carry out GO analysis are biased due to over-detection of differential expression for long and highly expressed transcripts. One tool that provides methods for performing GO analysis of RNA-Seq data while taking length bias into account is goseq. This tool needs 2 files as inputs -

* A tabular file with the differentially expressed genes from all genes assayed in the RNA-Seq experiment with 2 columns
  a. The Gene IDs (unique within the file), in uppercase letters
  b. A boolean indicating whether the gene is differentially expressed or not (True if differentially expressed or False if not)
* A file with information about the length of a gene to correct for potential length bias in differentially expressed genes

Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway database is a collection of pathway maps representing current knowledge of molecular interaction, recation and relation networks. A map can integrate many entities including genes, proteins, RNAs, chemical compounds, glycans, and chemical reactions, as well as disease genes and drug targets. Along with GO categories, including in KEGG pathway information enriches the understanding of what pathways the genes belong to.

GO pathway analysis
===================

At this stage, we have extracted genes that are significantly differentially expressed and visualized them. We are now ready to carry out Gene Ontology (GO) analysis using goseq.

For users that are running the workflow, these are the tools that would carry out a Gene Ontology analysis -
* Prepare the first dataset for goseq
  
  a. Compute
  
  b. Cut
  
  c. Change case

* Prepare the gene length file

  a. Extract dataset
  
  b. Change case
  
  c. goseq

For this step, workflow users don't need to change any parameters or enter any values. You can read further down this page if you would like to know how the other tools work and what they do (This will also show you the default parameters), or navigate to the next step -

For users who are running each step or those that are carrying out the secondary analysis seperately, let's start with the steps -

The first step is preparing the first dataset for "goseq" -

* The first tool to be used is "Compute"

* Under "Add expression", enter "bool(c7<0.05)"

* Select the output of DESeq2 - the DESeq2 result file under "as a new column to"

* Click on "Execute"

"Cut" tool is the second tool which will be used -

* Under "Cut columns", enter "c1,c8" (We want the gene name and true/false column)

* Select "tab" under "Delimited by"

* "From" the output of the "Compute" tool

* Click on "Execute"

The last tool in this step is "Change Case"

* Select the output of the "Cut" tool under "From"

* Under "Change case of columns", enter c1

* Use "tab" as the delimiter

* Enter "Upper case" under "To"

* Click on "Execute"

* Rename the output to "Gene IDs and differential expression"

The second step is preparing the gene lengths file -

* The first step under this is searching for the feature length collection generated by featureCounts from previous history and if run within this history, should be easily accessible

* The next tool that is used is "Extract Dataset"

* Select "featureCounts on collection N: Feature lengths" under "Input List"

* Under "How should a dataset be selected?", enter "The first dataset"

* Click on "Execute"

The last tool that will be used is "Change Case"

* Enter the output of "Extract Dataset" under "From"

* Under "Change case of columns", enter "c1"

* Use "tab" as the delimiter

* Enter "Upper case" under "To"

* Click on "Execute"

* Rename the output to "Gene IDs and length"

The last step in GO analysis is using the tool "goseq" to perfrom GO analysis

* Search for the tool under "Tools"

* Select the file "Gene IDs and differential expression" under "Differentially expressed genes file"

* Under "Gene lengths file", select "Gene IDs and length"

* Under "Gene categories", select "Get categories"

* In this category, select the genome to use under "Select a genome to use"

* Since Ensembl gene IDs were used, select "Ensembl Gene ID" under "Select Gene ID format"

* Select one or more categories - depending on which category you would want, select "GO: Cellular Component, GO: Biological Process, GO: Molecular Function"

* In "Output Options", select "Yes" to "Output Top GO terms plot"

* Select "Yes" to "Extract the DE genes for the categories (GO/KEGG terms)?"

With these parameters, "goseq" generates three outputs -

* A table (Ranked category list - Wallenius method) with the following columns for each GO term:
  
  a. category - GO category
  
  b. over_rep_pval - p-value for over-representation of the term in differentially expressed genes
  
  c. under_rep_pval - p-value for under-representation of the term in differentially expressed genes
  
  d. numDEInCat - number of differentially expressed genes in this category
  
  e. term - detail of the term
  
  f. ontology - MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)
  
  g. p.adjust.over_represented - p-value for over-representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure
  
  h. p.adjust.under_represented - p-value for under-representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure

* A graph with the top 10 over-represented GO terms

* A table with the differentially expressed genes associated to the GO terms

KEGG pathway analysis
=====================

For KEGG analysis, let's start with the users running the workflow. These are the tools that will be used -

* goseq

* Cut

* Pathview

For users using the workflow, you don't have to enter any parameters or values at this point. You can read further down this page if you would like to know how the other tools work and what they do (This will also show you the default parameters).

For users running each step seperately and for users running the secondary analysis seperately, let's start breaking down the steps to run a KEGG pathway analysis -

The first tool that we use is "goseq" which we used earlier -

* If you have performed the "Gene Ontology analysis" step, you would have the two files, "Gene IDs and differential expression" and "Gene IDs and length"
Under "Differentially expressed genes file", select "Gene IDs and differential expression" and select "Gene IDs and length" under "Gene lengths file"

* Select "Get categories" under "Gene categories" and the genome that you would like to select under "Select a genome to use"

* If your gene annotations contain Ensembl gene IDs, select "Ensembl Gene ID" under "Select Gene ID format"

* Under "Select one or more categories", pick "KEGG"

* In "Output options", select "No" under "Output Top GO terms plot"

* Under "Extract the DE genes for the categories (GO/KEGG terms)", select "Yes"

* Click on "Execute"

This tool outputs two tables -
* A large table with the KEGG terms and related statistics

* A table with the differentially expressed genes associated with the KEGG pathways 

The second tool that we use is "Cut" 

We have KEGG analysis done and we have diffferentially expressed gene's information and we would like to put together and overlay the log2 FC on the KEGG pathway

* A new table can be created from a earlier table using "Cut"

* Search for the tool under "Tools"

* Under "Cut columns", enter "c1, c3" (We want to extract the Gene ID and Log2Fold change for the genes that have a significant adjusted p-value)

* Select "Tab" under "Delimited by"

* The file "From" should be the earlier file we created "Genes with significant adj p-value"

* Rename the file to "Genes  with significant adj p-value and their Log2 FC"

* Create a new tabular file from the pathway IDs that you are interested in plotting and rename it as "KEGG pathways to plot"

The third tool that we will use is called "Pathview"

* Search for the tool under "Tools"

* Under "Number of pathways to plot", select "Multiple"

* Select "KEGG pathways to plot" under "KEGG pathways"

* Under "Does the file have header (a first line with column names)?", select "No" (Until your file has headers)

* Under species to use, select the genome you are working with

* Under "Provide a gene data file?", set "Yes"

* Enter the output of Cut under "Gene data"

* "Does the file have header (a first line with column names)?", select "No" 

* Under "Format for gene data", enter "Ensembl gene ID"

* Set "No" to "Provide a compound data file?"

* In "Output options"

* Select "KEGG native" for "Output for pathway"

* Set "Yes" for "Plot on same layer"

* Click on "Execute"
